src/AbstractAlgebra.jl

```diff
@@ -375,6 +375,12 @@ macro show_name(io, O)
 end
 
 function find_name(A, M = Main)
+  if M === Main && isdefined(Main, :CurrentModule)
+    a = find_name(A, Main.CurrentModule)
```

Comment:
One danger with this approach is that there is no `teardown` for `jldoctest` (though I've now submitted a requested that this should be added). So we can't easily unset `Main.CurrentModule`. Hopefully this won't matter in practice